pkg/lib/dpop/errors.go

```diff
@@ -0,0 +1,13 @@
+package dpop
+
+import "github.com/authgear/authgear-server/pkg/api/apierrors"
+
+var InvalidDPoPProof = apierrors.BadRequest.WithReason("InvalidDPoPProof")
+
+var ErrMalformedJwt = InvalidDPoPProof.New("malformed jwt")
+var ErrInvalidJwt = InvalidDPoPProof.New("invalid jwt")
+var ErrInvalidJwtPayload = InvalidDPoPProof.New("invalid jwt payload")
+var ErrInvalidJwtSignature = InvalidDPoPProof.New("invalid jwt signature")
+var ErrInvalidJwtNoJwkProvided = InvalidDPoPProof.New("jwk not provided in jwt")
+var ErrInvalidJwk = InvalidDPoPProof.New("invalid jwk")
+var ErrProofExpired = InvalidDPoPProof.New("proof expired")
```

Comment:
> We also want to use a distinct reason for each different cases, like DPoPMalformedJWT, DPoPInvalidJWTSignature, etc.

I used the same reason because the user basically handle all of them in the same way: Make the request again with a correct dpop jwt.